src/placeos-driver/interface/sensor.cr

```diff
@@ -0,0 +1,137 @@
+require "json"
+require "s2_cells"
+
+module PlaceOS::Driver::Interface::Sensor
+  enum SensorType
+    Temperature # https://en.wikipedia.org/wiki/Celsius
+    Humidity    # https://en.wikipedia.org/wiki/Humidity#Relative_humidity (percentage)
+    Illuminance # https://en.wikipedia.org/wiki/Illuminance (lux)
+    Pressure    # https://en.wikipedia.org/wiki/Pascal_(unit)
+    Trigger     # Proximity or motion etc
+    Switch      # On or Off - water leak, door open etc
+    Level       # Percentage + raw value + technical human readable (battery, fuel, water tank, oxygen, Co2)
+    Flow        # https://en.wikipedia.org/wiki/Cubic_metre_per_second (gas or liquid)
+    Counter
+    Acceleration # MetrePerSecondSquared
+    Speed        # MetrePerSecond
+    # Gyroscope: roll, pitch, yaw - Angle
+    Roll
+    Pitch
+    Yaw
+    Compass              # direction (magnetic field)
+    Current              # https://en.wikipedia.org/wiki/Ampere
+    Voltage              # https://en.wikipedia.org/wiki/Volt
+    ElectricalResistance # https://en.wikipedia.org/wiki/Ohm
+    Radiation            # https://en.wikipedia.org/wiki/Sievert
+    Distance
+  end
+
+  # Using SI units and SI derived units
+  enum Unit
```

Comment:
I think having a set of helpers to convert from common units would be useful. 
i.e. `farenheit_to_celsius` etc etc

Reply:
yeah makes sense to have a bunch of converter helpers in this interface